server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -430,7 +430,7 @@ public void messageReceived(final ForceSyncRequest request, TransportChannel cha
                 return;
             }
             startReplication(
-                ReplicationCheckpoint.empty(request.getShardId(), indexShard.getDefaultCodecName()),
+                ReplicationCheckpoint.empty(request.getShardId(), indexShard.getEngineCodec()),
```

Comment:
Can we refactor getLatestReplicationCheckpoint on IndexShard to return empty here if the shard is not yet active.  That would eliminate the need for another getEngineCodec method.

Reply:
Thanks for pointing this out. This also provides another opportunity to remove the `ReplicationCheckpoint` param from SegmentReplicationTarget as we are always fetch ReplicationCheckpoint from indexShard. Updating.